data/transactions/logic/sourcemap.go

```diff
@@ -0,0 +1,94 @@
+// Copyright (C) 2019-2022 Algorand, Inc.
+// This file is part of go-algorand
+//
+// go-algorand is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// go-algorand is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with go-algorand.  If not, see <https://www.gnu.org/licenses/>.
+
+package logic
+
+import (
+	"bytes"
+	"strings"
+)
+
+// sourceMapVersion is currently 3: https://sourcemaps.info/spec.html
+const sourceMapVersion = 3
+const b64table string = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/"
+
+// SourceMap contains details from the source to assembly process
+// currently contains map of TEAL source line to assembled bytecode position
+// and details about the template variables contained in the source file
+type SourceMap struct {
+	Version    int      `json:"version"`
+	File       string   `json:"file"`
+	SourceRoot string   `json:"sourceRoot"`
+	Sources    []string `json:"sources"`
+	Names      []string `json:"names"`
+	Mapping    string   `json:"mapping"`
+}
+
+// GetSourceMap returns a struct containing details about
+// the assembled file and encoded mappings to the source file.
+func GetSourceMap(sourceNames []string, offsetToLine map[int]int) SourceMap {
```

Comment:
@algorandskiy Reaching out to you for feedback on how to consider the following tradeoff:  Is it _safe_ to delete the following functionality or must we continue maintaining it? https://github.com/algorand/go-algorand/blob/376910dcdbe7ad2afb16a33fc5e785bb317e0296/cmd/tealdbg/debugger.go#L61-L62

_If_ we must maintain the functionality, then we'll update the PR to minimize/eliminate duplication.

Context:
* The question arises from a verbal with @algochoi.  We recognize the PR duplicates some tealdbg logic.
* In a future (beyond the PR) vision, we imagine the user supplying source maps to tealdbg.  If so, the future use case obviates the need for tealdbg to supply the source map.
  * For example, imagine having PyTeal-to-TEAL and TEAL-to-PC source maps.  Using both source maps enables a cleaner PyTeal debugging experience. 
*  tealdbg exposes source map and source via URLs dependent on session ID as shown below.  Since @algochoi and I are unfamiliar with _when_ a user may depend on the functionality and in our review it _felt_ like an obscure usage, we thought it prudent to see if removal is a reasonable option. 
 https://github.com/algorand/go-algorand/blob/376910dcdbe7ad2afb16a33fc5e785bb317e0296/cmd/tealdbg/cdtdbg.go#L76-L77

Reply:
We have the same conversation with @barnjamin.
My feedback is:
1. Sourcemap is a JS thing and IMO does not belong to logic package.
2. If the new functionality fully covers tealdbg needs it is fine to remove it from there.

The endpoints mentioned are used by CDT when debugging with sourcemap. It will not work without them.